launcher/src/main/java/org/apache/spark/launcher/SparkLauncher.java

```diff
@@ -82,8 +83,12 @@
   /** Used internally to create unique logger names. */
   private static final AtomicInteger COUNTER = new AtomicInteger();
 
+  public static final ThreadFactory REDIRECTOR_FACTORY = new NamedThreadFactory("launcher-proc-%d");
```

Comment:
It would be nice if the same factory were used by `ChildProcAppHandle` also. And this shouldn't be public.